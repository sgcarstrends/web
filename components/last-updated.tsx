import dayjs from "dayjs";

interface Props {
  lastUpdated: number;
}

export const LastUpdated = ({ lastUpdated }: Props) => (
  <div className="text-muted-foreground text-xs">
    Last updated:{" "}
    <span className="underline">
      {dayjs(lastUpdated).format("DD MMM YYYY, h:mmA")}
    </span>
  </div>
);
