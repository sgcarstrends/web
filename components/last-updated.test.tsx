import { render, screen } from "@testing-library/react";
import dayjs from "dayjs";
import { vi, describe, it, beforeAll, afterAll, expect } from "vitest";
import { LastUpdated } from "./last-updated";

const mockLastUpdated = 1735660800; // 1 Jan 2025, 00:00:00 GMT+8

describe("LastUpdated", () => {
  let dayjsFormatSpy: ReturnType<typeof vi.spyOn>;

  beforeAll(() => {
    // Mock dayjs format so tests aren't dependent on environment/timezone
    dayjsFormatSpy = vi
      .spyOn(dayjs.prototype, "format")
      .mockReturnValue("1 Jan 2025, 00:00AM");
  });

  afterAll(() => {
    dayjsFormatSpy.mockRestore();
  });

  it("should render", () => {
    const { container } = render(<LastUpdated lastUpdated={mockLastUpdated} />);
    expect(container).toMatchSnapshot();
  });

  it("renders the separator and label", () => {
    render(<LastUpdated lastUpdated={mockLastUpdated} />);
    expect(screen.getByText(/Last updated:/)).toBeVisible();
  });

  it("displays the mocked formatted date", () => {
    render(<LastUpdated lastUpdated={mockLastUpdated} />);
    // We expect our mock return value to appear in the DOM
    expect(screen.getByText("1 Jan 2025, 00:00AM")).toBeVisible();
  });

  it("formats date using dayjs with correct format string", () => {
    render(<LastUpdated lastUpdated={mockLastUpdated} />);
    // Verify dayjs format was called with the expected format string
    expect(dayjsFormatSpy).toHaveBeenCalledWith("DD MMM YYYY, h:mmA");
  });
});
