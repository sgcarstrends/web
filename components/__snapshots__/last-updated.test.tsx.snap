// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LastUpdated > should render 1`] = `
<div>
  <div
    class="text-muted-foreground text-xs"
  >
    Last updated:
     
    <span
      class="underline"
    >
      1 Jan 2025, 00:00AM
    </span>
  </div>
</div>
`;
